### ISSUE TITLE
Text that spans multiple elements

### ISSUE THREAD

**Issue by mathiasbynens** [tomayac/text-fragments-polyfill#1]
Reading https://github.com/tomayac/text-fragments-polyfill/blob/9c00c80ddd9a37dbf766fbc4d1524826211bd761/inject.js#L119-L121, it seems like this wouldn’t match the text `'foobar'` in e.g. `<b>foo</b>bar`.

One approach that could work is to walk the DOM, concatenate all text into a single string, then search that string for matches. For each match, you can then figure out which element(s) it corresponds to.

**Comment by tomayac** [comment]
Actually, independent of the incomplete implementation, this example will definitely [fail to match](https://wicg.github.io/ScrollToTextFragment/#example-73638554:~:text=%3A~%3Atext%3DThe%20quick%2Clazy%20dog,appear%20within%20a%20single%2C%20uninterrupted%20block.).

**Comment by mathiasbynens** [comment]
> Actually, independent of the incomplete implementation, this example will definitely [fail to match](https://wicg.github.io/ScrollToTextFragment/#example-73638554:~:text=%3A~%3Atext%3DThe%20quick%2Clazy%20dog,appear%20within%20a%20single%2C%20uninterrupted%20block.).

The section you link to talks about block boundaries, but [`<b>` is not a block-level element](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-b-element).

https://wicg.github.io/ScrollToTextFragment/#ref-for-concept-range②①:~:text=Collection%20breaks%20when%20we%20hit%20a,container)%20within%20a%20single%20block%2Dlevel%20container.

**Comment by tomayac** [comment]
You are absolutely right. Here's a test link that shows this working: https://pastoral-vagabond-beryl.glitch.me/#:~:text=https%3A%2F%2Fblog.chromium.org%2F2019%2F12%2Fchrome%2D80%2Dcontent%2Dindexing%2Des%2Dmodules.html%23HTML1.
